commands.c

```diff
@@ -91,14 +91,20 @@ void commands_process_packet(unsigned char *data, unsigned int len,
 	switch (packet_id) {
 	case COMM_FW_VERSION: {
 		int32_t ind = 0;
-		uint8_t send_buffer[50];
+		uint8_t send_buffer[24 + HW_NAME_MAX_LENGTH];
 		send_buffer[ind++] = COMM_FW_VERSION;
 		send_buffer[ind++] = FW_VERSION_MAJOR;
 		send_buffer[ind++] = FW_VERSION_MINOR;
 
-		strcpy((char*)(send_buffer + ind), HW_NAME);
+		strncpy((char*)(send_buffer + ind), HW_NAME, HW_NAME_MAX_LENGTH);
```

Comment:
strncpy will not null-terminate the string and strlen below will still advance ind past the length of the string. I think I would prefer to just increase the send buffer size to something like 256 and assume that no one makes the hardware name insanely long. Another way to limit the hardware name size is to copy the name into a variable at boot and terminate it if needed when it becomes too long, but this is also a case of adding a lot of complexity to cover a rare case that probably never occurs. In general you can be defensive to infinity with programming and still make mistakes, and at some point you have to ask yourself if the added complexity is worth the possible protection, or if it is more likely to even introduce new bugs that are larger problems (as is the case here).